### PR TITLE
Feat/derivatives fix command close potision

### PR DESCRIPTION
**Pull request by mkXultra**

### CODE REVIEW

File: scripts/setup/init.sh [hunk 1]
```diff
@@ -78,13 +79,17 @@ jq '.app_state.derivatives.params.perpetual_futures.commission_rate = "0.001"' $
 jq '.app_state.derivatives.params.perpetual_futures.margin_maintenance_rate = "0.5"' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
 jq '.app_state.derivatives.params.perpetual_futures.imaginary_funding_rate_proportional_coefficient = "0.0005"' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
 jq '.app_state.derivatives.params.perpetual_futures.markets = [{"base_denom": "ubtc", "quote_denom": "uusd" }]' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
```
Comment (taiki1frsh) [inline scripts/setup/init.sh:81]: please change "uusd" to "uusdc".
didn't check it, sorry.

File: scripts/commands/derivatives/msgs.sh [hunk 2]
```diff
@@ -12,8 +12,11 @@ $BINARY tx derivatives burn-lpt 1 ubtc --from=$USER1 $conf | jq .raw_log | sed '
 $BINARY tx derivatives burn-lpt 1 ubtc --from=$USER1 $conf | jq .raw_log | sed 's/\\n/\n/g'
 
 # open-position perpetual-futures
-$BINARY tx derivatives open-position perpetual-futures 100ubtc ubtc uusdc long --from=$USER1 $conf | jq .raw_log | sed 's/\\n/\n/g'
-$BINARY tx derivatives open-position perpetual-futures 100uusdc ubtc uusdc short --from=$USER1 $conf | jq .raw_log | sed 's/\\n/\n/g'
+$BINARY tx derivatives open-position perpetual-futures 100ubtc ubtc uusd long --from=$USER1 $conf | jq .raw_log | sed 's/\\n/\n/g'
```
Comment (taiki1frsh) [inline scripts/commands/derivatives/msgs.sh:15]: plz change "uusd" to "uusdc" after the change of the markets values in genesis.json.
plz refer init.sh file comment

File: x/derivatives/keeper/positions.go [hunk 3]
```diff
@@ -105,11 +124,21 @@ func (k Keeper) DeletePosition(ctx sdk.Context, address sdk.AccAddress, id strin
...
 func (k Keeper) OpenPosition(ctx sdk.Context, msg *types.MsgOpenPosition) error {
-	sender := msg.Sender.AccAddress()
-	lastPositionId := k.GetLastPositionId(ctx)
+	// todo check sender amount for margin
 
-	positionKey := types.AddressPositionWithIdKeyPrefix(sender, lastPositionId)
-	positionId := string(positionKey)
+	lastPosition, err := k.GetLastPosition(ctx)
+	if err != nil {
+		panic("failed to get last position")
```
Comment (taiki1frsh) [inline x/derivatives/keeper/positions.go:131]: maybe better to define Error case and return err, not panic?

File: scripts/commands/derivatives/msgs.sh (hunk 2 above)
Comment (mkXultra) [reply scripts/commands/derivatives/msgs.sh:15]: uusdc does't work.
uusd is working

File: x/derivatives/keeper/positions.go (hunk 3 above)
Comment (mkXultra) [reply x/derivatives/keeper/positions.go:131]: This case is usually not possible.
In such cases, Chain should be dropped.

Comment (mkXultra) [reply x/derivatives/keeper/positions.go]: Sorry, I made a mistake.
I've fixed it

File: scripts/setup/init.sh (hunk 1 above)
Comment (mkXultra) [reply scripts/setup/init.sh:81]: fixed

**Comment by mkXultra** [comment]
@taiki1frsh 
plz review again

File: scripts/commands/derivatives/msgs.sh (hunk 2 above)
Comment (taiki1frsh) [reply scripts/commands/derivatives/msgs.sh:15]: with this modification (https://github.com/UnUniFi/chain/pull/388#discussion_r1112528525), it should work.

Comment (mkXultra) [reply scripts/commands/derivatives/msgs.sh:15]: No, it does not work.
I have tried that.

File: scripts/setup/init.sh [hunk 4]
```diff
@@ -69,22 +69,27 @@ jq '.app_state.pricefeed.params.markets = [
...
 jq '.app_state.derivatives.params.perpetual_futures.margin_maintenance_rate = "0.5"' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
 jq '.app_state.derivatives.params.perpetual_futures.imaginary_funding_rate_proportional_coefficient = "0.0005"' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
-jq '.app_state.derivatives.params.perpetual_futures.markets = [{"base_denom": "ubtc", "quote_denom": "uusd" }]' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
-jq '.app_state.bank.denom_metadata = [{"base" : "ubtc" , "symbol": "ubtc"}, {"base" : "uusd", "symbol": "uusdc"}]' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
+jq '.app_state.derivatives.params.perpetual_futures.markets = [{"base_denom": "ubtc", "quote_denom": "uusdc" }]' $NODE_HOME/config/genesis.json > temp.json ; mv temp.json $NODE_HOME/config/genesis.json;
+jq '.app_state.bank.denom_metadata = [
+  {"base" : "ubtc" , "symbol": "ubtc"},
+  {"base" : "uusd", "symbol": "uusdc"}
```
Comment (taiki1frsh) [inline scripts/setup/init.sh:84]: plz change "uusd" to "uusdc" here

File: scripts/commands/derivatives/msgs.sh (hunk 2 above)
Comment (taiki1frsh) [reply scripts/commands/derivatives/msgs.sh:15]: sorry, needed one more modification.
https://github.com/UnUniFi/chain/pull/388#discussion_r1113750136

File: scripts/setup/init.sh (hunk 4 above)
Comment (mkXultra) [reply scripts/setup/init.sh]: fixed

File: scripts/commands/derivatives/msgs.sh (hunk 2 above)
Comment (mkXultra) [reply scripts/commands/derivatives/msgs.sh]: It's working

**Review by taiki1frsh — APPROVED** [review]

**Pull request merged**